src/ds3.c

```diff
@@ -1227,7 +1227,8 @@ ds3_error* ds3_head_object(const ds3_client* client, const ds3_request* request,
     ds3_string_multimap* return_headers;
     ds3_metadata* metadata = NULL;
 
-    if (num_chars_in_ds3_str(request->path, '/') < 2){
+    if (num_chars_in_ds3_str(request->path, '/') < 2
+       || '/' == request->path->value[request->path->size-1]) {
```

Comment:
Would this change mean you couldn't call head_object on an empty folder? Like an object called "directory/"?